doc/hooks-config.rst

```diff
@@ -0,0 +1,127 @@
+===============================
+Hook Configuration Options
+===============================
+
+As of version 4.4, |PyInstaller| implements a mechanism for passing
+configuration options to the hooks. At the time of writing, this
+feature is supported only in :ref:`.spec files <using spec files>` and
+has no command-line interface equivalent.
+
+The hook configuration options consist of a dictionary that is passed
+to the ``Analysis`` object via the ``hooksconfig`` argument. The keys
+of the dictionary represent `hook identifiers` while the values are
+dictionaries of hook-specific keys and values that correspond to
+hook settings:
+
+
+.. code-block:: python
+
+    a = Analysis(["program.py"],
+                ...,
+                hooksconfig={
+                    "hook1": {
+                        "option-list": ["entry1", "entry2"],
+                        "option-int": 42,
```

Comment:
The `option-list` and `option-int` kind of hints that there's a specific `option-[type]` format going on when there isn't - they key names are completely arbitrary. Maybe nonsense them up a bit?

Reply:
Or just `some-list-option` and `an-integer-option` would do.

Reply:
Good idea; I've randomized those example names a bit.